Add Chaibot test failure triage workflow to ci-chat-bot (#80476)
openshift/release: DEPLOY_CHAIBOT.md

### 2. Get OpenAI API Key

Option A - New Key:
1. Go to https://platform.openai.com/api-keys
2. Create new secret key
3. Copy the key (starts with `sk-`)
4. **Important**: Save it securely - you can't view it again

Option B - Use Existing:
If your organization already has a key in Vault, confirm the path.

### 3. Verify Cluster Access

```bash
# Login to app.ci cluster
oc login https://api.ci.l2s4.p1.openshiftapps.com:6443

# Verify access to ci namespace
oc get pods -n ci
```

## Deployment Steps

### Step 1: Update Slack Channel ID

```bash
# Edit the ConfigMap with the actual channel ID
vi clusters/app.ci/ci-chat-bot/chaibot-configmap.yaml

# Find this line (around line 12):
#   channel_id: "REPLACE_WITH_CHANNEL_ID"
# Replace with actual ID:
#   channel_id: "C01234ABCD"  # Your actual channel ID
```

### Step 2: Create Secret for OpenAI API Key

**Option A: Via kubectl (for testing/dev)**

```bash
# Create the secret directly
oc create secret generic ci-chat-bot-chaibot-secrets \
  --from-literal=openai-api-key="sk-YOUR-ACTUAL-KEY-HERE" \
  -n ci \
  --dry-run=client -o yaml | oc apply -f -

# Verify

> oc get secret ci-chat-bot-chaibot-secrets -n ci

Reviewer: The secret is managed by secretsync (from the vault item at `selfservice/cspi-qe/chaibot-openai-key`). No manual `oc create secret` is needed.

```suggestion
# The secret is managed by secretsync from the vault item.
# Verify it exists:
oc get secret cluster-secrets-chaibot-openai-key -n ci
```

```

**Option B: Via ci-secret-bootstrap (for production)**

```bash
# 1. Store the key in Vault (ask DPTP team for path)

# 2. Add to core-services/ci-secret-bootstrap/_config.yaml:
- from:
    openai-api-key:
      path: <vault-path-to-key>
  to:
    - cluster: app.ci
      namespace: ci

>       name: ci-chat-bot-chaibot-secrets

Reviewer: ```suggestion
      name: cluster-secrets-chaibot-openai-key
```

```bash
# Check pod status
oc get pods -n ci -l app=ci-chat-bot

# Check logs for Chaibot initialization
oc logs -n ci deployment/ci-chat-bot -c bot --tail=100 | grep -i chaibot

# Should see something like:
# INFO: Chaibot triage enabled
# INFO: Monitoring channels: [opp-discussion]
# INFO: AI provider: openai (model: gpt-4)
```

### Step 7: Test Functionality

**Method 1: Post a test message in Slack**

In `#opp-discussion`:
```
Test failure: https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/12345/...
```

Wait 30-60 seconds for Chaibot to respond in thread.

**Method 2: Check metrics**

```bash
# Port-forward to access metrics
oc port-forward -n ci deployment/ci-chat-bot 9090:9090 &

# Query metrics
curl http://localhost:9090/metrics | grep chaibot

# Look for:
# chaibot_messages_processed_total
# chaibot_failures_detected_total
# chaibot_analyses_completed_total
```

**Method 3: Monitor logs**

```bash
# Follow logs for Chaibot activity
oc logs -n ci deployment/ci-chat-bot -c bot -f | grep -i "chaibot\|triage"
```

## Troubleshooting

### Pod won't start

```bash
# Check events
oc describe pod -n ci -l app=ci-chat-bot

# Common issues:

> # - Missing secret: ci-chat-bot-chaibot-secrets

Reviewer: ```suggestion
# - Missing secret: cluster-secrets-chaibot-openai-key
```
